File: Sample/Sample/TabBarController.m
/* Copyright Airship and Contributors */

#import "TabBarController.h"

@implementation TabBarController

#ifdef DEBUG
- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Instantiate debug library and add it to the tab bar
    
    // Instantiate debug library storyboard
    UIStoryboard *debugStoryboard = [UIStoryboard storyboardWithName:@"AirshipDebug" bundle:[NSBundle bundleWithIdentifier:@"com.urbanairship.AirshipDebug"]];
    if (!debugStoryboard) {
        return;
    }
    
    // Instantiate debug library's initial view controller
    UIViewController *debugViewController = [debugStoryboard instantiateInitialViewController];
    if (!debugViewController) {
        return;
    }
    
    // Debug library needs to operate in a nav controller
    UINavigationController *debugNavController = [[UINavigationController alloc] initWithRootViewController:debugViewController];
    if (!debugNavController) {
        return;
    }
    
    // Create new tab bar item for debug library
    UITabBarItem *debugItem = [[UITabBarItem alloc] initWithTitle:@"Debug" image:[UIImage imageNamed:@"outline_bug_report_black_36pt"] tag:0];
    if (!debugItem) {
        return;
    }
    debugNavController.tabBarItem = debugItem;
    
    // Add tab to tab bar
    NSMutableArray *viewControllers = [NSMutableArray arrayWithArray:self.viewControllers];
    [viewControllers addObject:debugNavController];
    self.viewControllers = viewControllers;
}
#endif

@end

File: Sample/Sample/TabBarController.swift
/* Copyright Airship and Contributors */

import UIKit
import SwiftUI
import AirshipDebug

class TabBarController: UITabBarController {
    #if DEBUG
    override func viewDidLoad() {
        super.viewDidLoad()

        let debugViewController = UIHostingController(
            rootView: AirshipDebugView()
        )
        
        // Debug library needs to operate in a nav controller
        let debugNavController = UINavigationController(rootViewController: debugViewController)
        
        // Create new tab bar item for debug library
        let debugTitle = NSLocalizedString("ua_debug_tab_title", tableName: "UAPushUI", comment: "Debug")
        let debugItem = UITabBarItem.init(title: debugTitle, image: UIImage.init(named: "outline_bug_report_black_36pt"), tag: 0)
        debugNavController.tabBarItem = debugItem
        
        // Add tab to tab bar
        viewControllers?.append(debugNavController)
    }
    #endif
}
